### PR TITLE
Add a debug mode for troubleshooting

### PR DESCRIPTION
**Pull request by michaelrambeau**
## Goal

In order to help when troubleshooting issues in the Merge CLI, this PR introduces the "debug" mode.

What "debug" mode allows you to see:

- Every HTTP request made by the CLI, with the end-point called, the response code and the time it takes to get the response
- Every call to Webpack with the actual version, the input file (the entry) and the generated output with its size (the bundle), and the time it took
- Every serialization process for every single component

It relies on the well-known (a kind of standard in the Node.js community) `debug` package: https://github.com/debug-js/debug

You add an environment variable called `DEBUG` in the command line with the "scope" of the debug statements you want to see, so you can adjust to inspect only things you are interested in.

## How to use it

All you need is to prefix the command (push, experimental mode... anything) with a `DEBUG` environment variable.

Example:

```
DEBUG=uxpin* npx uxpin-merge push --token ***
```

All output is under the namespace `uxpin`.

Then there are sub-parts: `uxpin:http`, `uxpin:webpack`
So basically you start the command with `DEBUG=uxpin*` to get **everything** related to UXPin Merge.

if you do  `DEBUG=uxpin` you'd get only the **top level uxpin entries** (which only gives a raw overview of the process).

Don't try `DEBUG=*` because Babel also uses it and it prints a lot of things (every single statement parsed, pages and pages of code...) don't do that! 😄 

## Screenshot

image

### CODE REVIEW

**Comment by Evomatic** [comment]
It works well! Hopefully with this kind of insight, we will be able to debug the CLI a bit better.
-zsh 2023-06-30 at 10 25 03 AM

File: packages/uxpin-merge-cli/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@uxpin/merge-cli",
-  "version": "3.0.5",
+  "version": "3.0.5-debug-mode",
```
Comment (Evomatic) [inline packages/uxpin-merge-cli/package.json:3]: Please correct me if I am wrong, but should `-debug-mode` be removed from the version? If we plan to release this as a latest version of the CLI?

**Review by Evomatic — COMMENTED** [review]
Looks good overall! I have one comment!

**Comment by michaelrambeau** [comment]
Adding a commit to get more visibility when the function `isGitRepository`... and to try to update this PR that does not reflect the latest commits pushed from the repo!!!

In debug mode, the failure message would be displayed.

So, if there is no Git repo, the following should be displayed: `fatal: not a git repository (or any of the parent directories): .git`

image

File: packages/uxpin-merge-cli/src/steps/serialization/vcs/repositories/git/util/isGitRepository.ts [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
 
+const log = debug('uxpin');
+
 export async function isGitRepository(cwd: string): Promise<boolean> {
   try {
     await execAsync('git status', { cwd });
     return true;
-  } catch (e) {
+  } catch (error) {
+    log(`git status failed`, (error as Error).message);
```
Comment (Evomatic) [inline packages/uxpin-merge-cli/src/steps/serialization/vcs/repositories/git/util/isGitRepository.ts:12]: Can we maybe remove the `git status failed`? Right now when we get the error message, it says

`git status failed Command failed: git status` which almost sounds a bit repetitive. This sounds a bit more precise and easy to follow `Command failed: git status`

What do you think?

**Review by Evomatic — CHANGES_REQUESTED** [review]
I left one comment, let me know when you think!

File: packages/uxpin-merge-cli/src/steps/serialization/vcs/repositories/git/util/isGitRepository.ts (hunk 2 above)
Comment (michaelrambeau) [reply packages/uxpin-merge-cli/src/steps/serialization/vcs/repositories/git/util/isGitRepository.ts:12]: I wanted an error message that gives the context it's not the Git command itself that breaks, but the `execAsync` used to run a shell from Node.js.

Something like "Unable to call Git command:", followed by the error message would be better?

Comment (Evomatic) [reply packages/uxpin-merge-cli/src/steps/serialization/vcs/repositories/git/util/isGitRepository.ts:12]: Ok I understand now, thank you for explaining! In that case, for context I think `Unable to call Git command:` sounds better. Gives better context. 👍

Comment (michaelrambeau) [reply packages/uxpin-merge-cli/src/steps/serialization/vcs/repositories/git/util/isGitRepository.ts:12]: Thinking more , we have already a final `ERROR Could not determine...` at the end, always printed (whether it's debug or not), so no need to be too verbose, you are right.

So we can do this:

- Show a message before the check
- Show only the raw error message if it fails, like this:

image

**Review by Evomatic — APPROVED** [review]
Thank you for the work, LGTM! 🧑‍💻
